File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i in range(len(nums)):
            if (target - nums[i]) in m:
                return [i, m[target - nums[i]]]
            m[nums[i]] = i
        return [-1, -1]

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> s = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (s.containsKey(target - nums[i])) {
                return new int[]{i, s.get(target - nums[i])};
            }
            s.put(nums[i], i);
        }
        return new int[]{-1, -1};
    }
}